### PR TITLE
NanoUI 3.0!

### LINKED ISSUE #13460: Nano UI is not potato computer compatible
d10_icon

One of the great things about SS13 is that it can be run on machines that are absolutely not suited for gaming. Calling out for all the spessmen with poverty tier computers here. It stings pretty bad that some people are probably getting shut out of a fairly critical part of the game due to this. 

Is there anything that can be done to give the affected people some sort of fallback for these things, even if it doesn't look quite so slick?

### LINKED ISSUE #13641: NanoUI for gas cannisters is shit
Cannister UI is so trash, it's impossible to use at any speed except 'excruciatingly slow'. Twice in one shift now I've gone to fill plasma tanks, put the tank in, waited for it to fill, pressed close, waited for it to say close, then pressed eject, only for the cannister to suddenly undo it's 'close' and then eject the tank and spill fucking plasma all over engineering.

How am I supposed to rely on this without waiting 5000 light years to make sure it actually closes when the UI says it closes before ejecting?

### PR DESCRIPTION
**Pull request by neersighted**
A complete rewrite of the NanoUI frontend to be more modern, and yet still run on potatoes (XP/IE8).

Note that this makes breaking changes to template syntax, and introduces new build tools.

This resolves #13460.
This resolves #13641.

Also, snuck in a stealth-change, minimaps are now stored per-map, meaning we don't have to regenerate them on map change.

Demo

### CODE REVIEW

**Comment by neersighted** [comment]
Thanks to @tkdrg for testing on Windows XP/IE8 and WINE.

**Comment by optimumtact** [comment]
looks good to me

**Comment by WJohn** [comment]
30k lines of code removed and 105 file changes? Did you screw something up or were there really that many changes, because that's insane.

**Comment by neersighted** [comment]
Yep, it really changes/removes that much.

**Comment by Dorsisdwarf** [comment]
There's a lot of whitespace and CSS being removed, which boosts the removal count

**Comment by Jalleo** [comment]
@neersighted does it recognise if the map has been updated sincelast run or not?

**Comment by neersighted** [comment]
Indeed it does, using md5sums.

**Comment by duncathan** [comment]
>30,000 deletions

YOU BEAUTIFUL BEAUTIFUL PERSON

**Comment by ike709** [comment]
>+1,333 −30,528 for a presumably better system

And the Coder of the Day award goes to...

**Comment by optimumtact** [comment]
guys they just removed all the scripts and made them load from the vendor provided cdn's

**Comment by Lo6a4evskiy** [comment]
Yeah, lines of code is a bad way to measure anything.

**Comment by Dorsisdwarf** [comment]
It removes 30,000 lines of code that have to be compiled and downloaded and shit

**Comment by Shadowlight213** [comment]
Neersighted for best coder 2015

**Comment by Dorsisdwarf** [comment]
soah many commit

**Comment by Shadowlight213** [comment]
@neersighted yeah, you're probably going to want to squash your commits at some point.

**Comment by neersighted** [comment]
That is the plan... This is still very much WIP. I don't usually shitcommit :p

**Comment by neersighted** [comment]
PR updated with preview of new Air Alarm.

**Comment by Razharas** [comment]
Ask for live testing on servers again

**Comment by neersighted** [comment]
Redesigned some stuff, here's what it looks like now:

Demo

**Comment by bear1ake** [comment]
It's happening?

**Comment by optimumtact** [comment]
Soon

**Comment by neersighted** [comment]
I think this is now ready to testmerge. If the testmerge goes well I will squash my commits and be ready to merge.

**Comment by KorPhaeron** [comment]
We can not test merge this because it involves preferences and will mess up save files

At least I was told as much for hand of god

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -771,81 +771,72 @@
...
+			if(usr.has_unlimited_silicon_privilege)
+				if(emagged || (stat & (BROKEN|MAINT)))
+					usr << "The APC does not respond to the command."
+				else
+					locked = !locked
+					update_icon()
+		if("overload")
+			if(usr.has_unlimited_silicon_privilege)
+				src.overload_lighting()
+		if("hack")
```
Comment (KorPhaeron) [inline code/modules/power/apc.dm]: This is the option I want to show up for traitor AI's

Do you know where I go to change what shows up on the list?

Comment (neersighted) [inline code/modules/power/apc.dm]: You need to edit the malf status proc to make malf options show up for traitor AIs.

**Pull request merged**